### ISSUE TITLE
shopify deploy heroku - Could not find any git branches

### ISSUE THREAD

**Issue by enginestudios** [Shopify/shopify-app-cli#887]
# Issue summary

On a fresh shopify cli project, when shopify deploy heroku command is run, this error appears

`Could not find any git branches`

However I am still able to deploy to heroku using `git push heroku master`, but the app produces this error when it is installed in a store
```
Error {name: "AppBridgeError", message: "APP::ERROR::INVALID_CONFIG: apiKey must be provided", action: undefined, type: "APP::ERROR::INVALID_CONFIG", stack: "AppBridgeError: APP::ERROR::INVALID_CONFIG: apiKey must be provided"}
```

adding this in the package.json and using `git push heroku master` again solves the problem
 
```
 "engines": {
    "node": "10.x"
  },
```

## Expected behavior

Deploy to heroku should be successful when `shopify deploy heroku` is run from the command line

## Actual behavior

Error message appears
`Could not find any git branches`

![Screenshot-2020-09-28-at-5 29 36-PM](https://user-images.githubusercontent.com/8068983/94417361-90cb9680-01b2-11eb-8f27-08b15748397d.jpg)

## Steps to reproduce the problem

1.  `git init`
2. `git add .`
3. `git commit -am "initial commit" `
4.  `shopify deploy heroku`

## Specifications

- App type:
- Operating System: Mac OS
- Shell: bash
- Ruby version (ruby -v): ruby 2.7.0p0

**Comment by paulomarg** [comment]
Thanks for reporting this, we will look into it. Just so we have the full information, can you please confirm your node version?

**Comment by enginestudios** [comment]
> Thanks for reporting this, we will look into it. Just so we have the full information, can you please confirm your node version?

thanks for looking into it! it's node v10.22.1

**Comment by thecodepixi** [comment]
Hi @enginestudios 👋  
I'm currently working on this issue so we can find a fix for you, and having trouble reproducing your exact error. I was hoping I could get more context on what you're working on to see if that helps us uncover what might be causing the error. 

You mentioned you're working on "a fresh shopify cli project", and I was curious if you could tell us the setup options you chose? (Rails/Node, database option, public/private app, etc). 

The code that throws the error you're getting uses `git branch --list` to check your local repo and see which branch to use for the deploy (and lets you choose a branch, if it finds more than one). Can you see what that command (`git branch --list`) returns from inside your app project directory? 

Thanks!

**Comment by enginestudios** [comment]
> Hi @enginestudios 👋
> I'm currently working on this issue so we can find a fix for you, and having trouble reproducing your exact error. I was hoping I could get more context on what you're working on to see if that helps us uncover what might be causing the error.
> 
> You mentioned you're working on "a fresh shopify cli project", and I was curious if you could tell us the setup options you chose? (Rails/Node, database option, public/private app, etc).
> 
> The code that throws the error you're getting uses `git branch --list` to check your local repo and see which branch to use for the deploy (and lets you choose a branch, if it finds more than one). Can you see what that command (`git branch --list`) returns from inside your app project directory?
> 
> Thanks!

Basically I followed the instructions here, and chose Nodejs
https://shopify.github.io/shopify-app-cli/app/node/

then after initialising git, and adding a commit, I ran the command `shopify deploy heroku`

my git branch is master

![Screenshot 2020-10-05 at 12 30 38 PM](https://user-images.githubusercontent.com/8068983/95040651-d9310a00-0706-11eb-9d3b-0518e082c277.png)

**Comment by andyw8** [comment]
@enginestudios could you please show the output of

`git --version`

and

`git branch --list --format='%(refname:short)'`

**Comment by enginestudios** [comment]
hi @andyw8 
git version 2.10.0

and i get an error for `git branch --list --format='%(refname:short)'`

![Screenshot 2020-10-06 at 10 02 10 AM](https://user-images.githubusercontent.com/8068983/95150582-122ab680-07bb-11eb-8e83-c82200de8ab9.png)

**Comment by andyw8** [comment]
@enginestudios We discovered that the `refname:short` format, which shopify-app-cli uses internally, is only available since git 2.13. Updating to a newer version should resolve this (version 2.10.0 was [released](https://github.com/git/git/releases/tag/v2.10.0) over 4 years ago).

**Comment by enginestudios** [comment]
thank you! I can confirm `shopify deploy heroku` works now after upgrading to git version 2.28.0
